components/action-mgt/org.wso2.carbon.identity.action.management/src/test/java/org/wso2/carbon/identity/action/management/util/ActionValidatorTest.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.action.management.util;
+
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import org.wso2.carbon.identity.action.management.exception.ActionMgtClientException;
+
+/**
+ * Unit test class for ActionValidator class.
+ */
+public class ActionValidatorTest {
+
+    private ActionValidator actionValidator;
+
+    @BeforeClass
+    public void setUp() {
+
+        actionValidator = new ActionValidator();
+    }
+
+    @AfterClass
+    public void tearDown() {
+
+        actionValidator = null;
+    }
+
+    @DataProvider
+    public Object[][] isBlankDataProvider() {
+
+        return new String[][]{
+                {"Action name", null},
+                {"Endpoint authentication URI", ""},
+                {"Endpoint authentication type", "  "},
+                {"Username", null},
+                {"Password", ""},
+                {"Access token", null},
+                {"API key header name", null},
+                {"API key value", null}
+        };
+    }
+
+    @Test(expectedExceptions = ActionMgtClientException.class, dataProvider = "isBlankDataProvider")
+    public void testIsBlank(String fieldName, String fieldValue) throws ActionMgtClientException {
+
+        actionValidator.isBlank(fieldName, fieldValue);
+    }
+
+    @DataProvider
+    public Object[][] isNotBlankDataProvider() {
+
+        return new String[][]{
+                {"Action name", "test-action"},
+                {"Endpoint authentication URI", "https://testapi.com"},
+                {"Endpoint authentication type", "BASIC"},
+                {"Username", "testuser"},
+                {"Password", "test@123"}
```

Comment:
Same as above. Testing one entry would be enough

Reply:
Updated with 487a8d524e3dd6f8618595b7b9d9266bd19f8c59